---
title: "Protein Coding"
output: html_notebook
---
Load in data
```{r}
ostsig <- read.csv(file.choose(), row.names = 1, stringsAsFactors = T)
yoda1 <- read.csv(file.choose(), row.names = 1, stringsAsFactors = T)
```

Filter osteocyte signature and osteocyte transcriptome (Youlten et al, 2021) for only protein-coding genes.

```{r}
ostsig_filter <- ostsig %>% filter(Biotype == 'protein_coding')
```
Done- now to remove NAs
```{r}
ostsig_filter <- ostsig_filter %>% filter(!(Human_Ensembl_ID == 'NA'))
```
All NAs removed- there are 1042 named protein coding genes in the osteocyte signature
Now to do the same to the transcriptome as a whole
```{r}
transcriptome_filter <- transcriptome %>% filter(Biotype == 'protein_coding')
transcriptome_filter <- transcriptome_filter %>% filter(!(Human_Ensembl_ID == 'NA'))
```
There are 18309 named protein coding genes in the osteocyte transcriptome
Compare Yoda1 data to both of the generated datasets
```{r}
yoda1_and_ostsig <- yoda1[yoda1$ensembl_ID %in% ostsig_filter$Human_Ensembl_ID,]
write.csv(yoda1_and_ostsig, "C:/Users/hamay/OneDrive - Cardiff University/R Stuff/spreadsheets/yoda1_and_ostsig.csv")
yoda1_and_transcriptome <- transcriptome_filter[transcriptome_filter$Human_Ensembl_ID %in% yoda1$ensembl_ID,]
```
Shared between Yoda1 and osteocyte signature = 539
Shared between Yoda1 and osteocyte transcriptome = 8630
Create unshared datasets
```{r}
yoda1_not_ostsig <- yoda1[yoda1$ensembl_ID %notin% ostsig_filter$Human_Ensembl_ID,]
ostsig_not_yoda1 <- ostsig_filter[ostsig_filter$Human_Ensembl_ID %notin% yoda1$ensembl_ID,]
yoda1_not_transcriptome <- yoda1[yoda1$ensembl_ID %notin% transcriptome_filter$Human_Ensembl_ID,]
transcriptome_not_yoda1 <- transcriptome_filter[transcriptome_filter$Human_Ensembl_ID %notin% yoda1$ensembl_ID,]
```
Genes in yoda1 and not the osteocyte signature = 8794
Genes in yoda1 and not the osteocyte transcriptome = 987
Genes in the osteocyte signature and not yoda1 = 503
Genes in the osteocyte transcriptome and not yoda1 = 9679
Which osteocyte signature and yoda1 shared genes are significantly regulated by Yoda1?
```{r}
yoda1ostsignif<-yoda1_and_ostsig[yoda1_and_ostsig$padj <= 0.05 & !is.na(yoda1_and_ostsig$padj),]
```
272 significant p-adjust values. What do these genes do?
```{r}
YODA1OSTSIG_go <- enricher(yoda1ostsignif$ensembl_ID, universe = yoda1_and_ostsig$ensembl_ID, 
             TERM2GENE=go[,c(4,1)], TERM2NAME=go[,c(4,5)])
nrow(YODA1OSTSIG_go@result[YODA1OSTSIG_go@result$p.adjust <= 0.05,])
barplot(YODA1OSTSIG_go, showCategory = 15, font.size = 7)
```
Extracellular region, collagen-containing ECM, endoplasmic reticulum lumen, and ECM. I'll remove these significant genes from the Yoda1 dataset since they are correlated with the osteocyte signature. The genes may skew the rest of the results if not. 
```{r}
yoda1_filter <- yoda1[yoda1$ensembl_ID %notin% yoda1ostsignif$ensembl_ID,]
```
9061 remaining genes which are not significantly correlated with the osteocyte genotype of the cells. Regulation of these will be due to Yoda1 treatment.
What are the unshared osteocyte signature genes doing?
```{r}
ostsignoyoda_signif<-ostsig_not_yoda1[ostsig_not_yoda1$Skeletal_maturation_cluster_corr_pval <= 0.05 & !is.na(ostsig_not_yoda1$Skeletal_maturation_cluster_corr_pval),]
OSTSIGNOYODA_go <- enricher(ostsignoyoda_signif$Human_Ensembl_ID, universe = ostsig_not_yoda1$ensembl_ID, 
             TERM2GENE=go[,c(4,1)], TERM2NAME=go[,c(4,5)])
nrow(OSTSIGNOYODA_go@result[YODA1OSTSIG_go@result$p.adjust <= 0.05,])
barplot(OSTSIGNOYODA_go, showCategory = 15, font.size = 7)
```
ECM, cell adhesion, collagen-containing ECM, skeletal system development. Clearly some important genes were missed out but that can only be expected with a dataset the size of our Yoda1 dataframe. Around 52% of the osteocyte signature genes can be found in our Yoda1 data, which is not a bad success rate.
How are the shared yoda1 and osteocyte signature genes regulated?
```{r}
yoda1_and_ostnotsig_up <- yoda1_and_ostsig %>% filter(log2FoldChange > 0) #198
yoda1_and_ostnotsig_down <- yoda1_and_ostsig %>% filter(log2FoldChange < 0) #341
```

How many of the shared osteocyte transcriptome and Yoda1 genes are significantly regulated?
```{r}
yoda1_and_transcriptome_sig<-yoda1_and_transcriptome[yoda1_and_transcriptome$Skeletal_maturation_cluster_corr_pval <= 0.05 & !is.na(yoda1_and_transcriptome$Skeletal_maturation_cluster_corr_pval),]
```
How many of the newly filtered genes in the Yoda1 dataset are significantly regulated?
```{r}
yoda1_filter_sig<-yoda1_filter[yoda1_filter$padj <= 0.05 & !is.na(yoda1_filter$padj),]
```
Which of the significantly regulated osteocyte signature and yoda1 genes are upregulated and how many are downregulated?
```{r}
yoda1_and_ostsig_up <- yoda1ostsignif %>% filter(log2FoldChange > 0)
yoda1_and_ostsig_down <- yoda1ostsignif %>% filter(log2FoldChange < 0)
write.csv(yoda1_and_ostsig_up, "C:/Users/hamay/OneDrive - Cardiff University/R Stuff/spreadsheets/yoda1_and_ostsig_up.csv")
write.csv(yoda1_and_ostsig_down, "C:/Users/hamay/OneDrive - Cardiff University/R Stuff/spreadsheets/yoda1_and_ostsig_down.csv")
yoda1_and_ostsig_up <- read.csv(file.choose(), row.names = 1, stringsAsFactors = T)
yoda1_and_ostsig_down <- read.csv(file.choose(), row.names = 1, stringsAsFactors = T)
```
103 are upregulated, 169 are downregulated.
Explore these datasets using gProfiler. Downregulated first.
```{r}
install.packages("gprofiler2")
library(gprofiler2)
??gprofiler2
gpro <- gost(yoda1_and_ostsig_down$ensembl_ID, organism = "hsapiens", ordered_query = FALSE, multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05)
gostplot(gpro, capped = TRUE, interactive = TRUE) #interactive plot
p <- gostplot(gpro, capped = FALSE, interactive = FALSE) #not interactive, uncapped
p
```
Then upregulated.
```{r}
gproup <- gost(yoda1_and_ostsig_up$ensembl_ID, organism = "hsapiens", ordered_query = FALSE, multi_query = FALSE, significant = TRUE, exclude_iea = FALSE, measure_underrepresentation = FALSE, evcodes = FALSE, user_threshold = 0.05)
gostplot(gproup, capped = TRUE, interactive = TRUE)
```
Separate gProfiler analyses of the datasets then performed externally.




